### DISCUSSION TITLE
Possible to include code files with syntax highlighting?

### DISCUSSION THREAD

**Discussion by matthewfeickert** [executablebooks/meta#1026] [Q&A]
I might be getting confused about the differences between what https://github.com/executablebooks/MyST-Parser is capable of now and what https://github.com/executablebooks/mystjs has implemented, but while following and trying to expand the [MyST Websites quickstart guide](https://myst-tools.org/docs/mystjs/quickstart-myst-websites) (https://github.com/executablebooks/mystjs-quickstart/tree/ac5b2afa5b560afd7809a25ad8e4c3bc002ca13f) I am trying to add my own example page that would demo the inclusion of code in static files but with relevant syntax highlighting:

````markdown
# Trying out codeblocks

## Python codeblock

```{code-block} python
:name: my-program
:caption: Creating a TensorMesh using SimPEG
:linenos:
:emphasize-lines: 3,4
from discretize import TensorMesh

hx = [(1, 40)]
hy = [(1, 40)]

mesh = TensorMesh([hx, hy])
```

## Now try with inserted file

```{include} code/example.py
```
````

```console
$ cat code/example.py
import numpy as np
from discretize import TensorMesh

hx = [(1, 40)]
hy = [(1, 40)]

mesh = TensorMesh([hx, hy])
```

This of course includes `code/example.py` but it is not rendered correctly as Python code.

![Screenshot from 2023-06-08 16-55-11](https://github.com/executablebooks/meta/assets/5142394/af000d97-426f-44fd-bd2e-a5f7c2da339a)

From the [myst-parser FAQ](https://myst-parser.readthedocs.io/en/latest/faq/index.html#include-a-file-from-outside-the-docs-folder-like-readme-md) it seems like [this should be possible to do](https://github.com/executablebooks/MyST-Parser/blob/80b892ff67734cc8e243c27c0e3dc0eceaeb5c71/docs/faq/index.md?plain=1#L69-L95) with

````markdown
```{literalinclude} code/example.py
:language: python
```
````

but including this produces the warning

> ⚠️  03-code-example.md:24 unknown directive: literalinclude

at `myst build`.

So is this something that is new enough to be included in the `myst-parser` spec but not yet implemented in `mystjs`?

## Environment details

```console
$ node --version
v18.16.0
$ npm --version
9.5.1
$ npm info myst-cli version
0.1.31
```

## Context for question

I am trying to understand if `mystjs` is something that I should look at for a scientific computing focused blog compared to Hugo. If the sort of things that I'm trying to do are a better fit for Jupyter Book than `mystjs` that would be helpful to know too!

**Comment by rowanc1** [comment]
Correct, the `literalinclude` directive is not yet implemented -- a mostly complete list of where `mystjs` is not yet up to spec with sphinx/jupyter-book is here:
https://github.com/executablebooks/mystjs/issues/189

I have moved `literalinclude` to the top of the list! We already support `include` so it should be pretty simple! I will circle back here when we get it done!

**Reply by matthewfeickert** [reply]
Excellent. Many thanks for the fast response, @rowanc1!

**Reply by rowanc1** [reply]
There is now a PR implementing `literalinclude` here: https://github.com/executablebooks/mystmd/pull/610 🚀
